src/common/colorspaces_inline_conversions.h

```diff
@@ -421,6 +422,125 @@ static inline float dt_camera_rgb_luminance(const float *const rgb)
+    LMS[i] = powf((c1 + c2 * LMS[i]) / (1.0f + c3 * LMS[i]), p);
+  }
+
+  // L'M'S' -> Izazbz
+  for(int i = 0; i < 3; i++) JzAzBz[i] = A[i][0] * LMS[0] + A[i][1] * LMS[1] + A[i][2] * LMS[2];
+  // Iz -> Jz
+  JzAzBz[0] = ((1.0f + d) * JzAzBz[0]) / (1.0f + d * JzAzBz[0]) - d0;
+}
+
+#ifdef _OPENMP
+#pragma omp declare simd
+#endif
+static inline void dt_JzAzBz_2_JzCzhz(const float *const JzAzBz, float *const JzCzhz)
+{
+  float var_H = atan2f(JzAzBz[2], JzAzBz[1]) / (2.0f * DT_M_PI_F);
+  JzCzhz[0] = JzAzBz[0];
+  JzCzhz[1] = sqrtf(JzAzBz[1] * JzAzBz[1] + JzAzBz[2] * JzAzBz[2]);
+  JzCzhz[2] = var_H >= 0.0f ? var_H : 1.0f + var_H;
+}
+
+#ifdef _OPENMP
+#pragma omp declare simd
+#endif
+static inline void dt_JzCzhz_2_JzAzBz(const float *const JzCzhz, float *const JzAzBz)
+{
+  JzAzBz[0] = JzCzhz[0];
+  JzAzBz[1] = cosf(2.0f * DT_M_PI_F * JzCzhz[2]) * JzCzhz[1];
+  JzAzBz[2] = sinf(2.0f * DT_M_PI_F * JzCzhz[2]) * JzCzhz[1];
+}
+
+#ifdef _OPENMP
+#pragma omp declare simd
+#endif
+static inline void dt_JzAzBz_2_XYZ(const float *const JzAzBz, float *const XYZ_D65)
+{
+  static const float b = 1.15f;
+  static const float g = 0.66f;
+  static const float c1 = 0.8359375f; // 3424 / 2^12
+  static const float c2 = 18.8515625f; // 2413 / 2^7
+  static const float c3 = 18.6875f; // 2392 / 2^7
+  static const float n = 0.159301758f; // 2610 / 2^14
+  static const float p = 134.034375f; // 1.7 x 2523 / 2^5
+  static const float d = -0.56f;
+  static const float d0 = 1.6295499532821566e-11f;
+  static const float MI[3][4] DT_ALIGNED_ARRAY = {
+      {  1.9242264357876067, -1.0047923125953657,  0.0376514040306180f, 0.0f },
+      {  0.3503167620949991,  0.7264811939316552, -0.0653844229480850f, 0.0f },
+      { -0.0909828109828475, -0.3127282905230739,  1.5227665613052603f, 0.0f },
+  };
+  static const float AI[3][4] DT_ALIGNED_ARRAY = {
+      {  1.0f,  0.1386050432715393f,  0.0580473161561189f, 0.0f },
+      {  1.0f, -0.1386050432715393f, -0.0580473161561189f, 0.0f },
+      {  1.0f, -0.0960192420263190f, -0.8118918960560390f, 0.0f },
+  };
+
+  float XYZ[4] DT_ALIGNED_PIXEL = { 0.0f, 0.0f, 0.0f, 0.0f };
+  float LMS[4] DT_ALIGNED_PIXEL = { 0.0f, 0.0f, 0.0f, 0.0f };
+  float IzAzBz[4] DT_ALIGNED_PIXEL = { 0.0f, 0.0f, 0.0f, 0.0f };
+
+  IzAzBz[0] = JzAzBz[0] + d0;
+  IzAzBz[0] = IzAzBz[0] / (1.0f + d - d * IzAzBz[0]);
+  IzAzBz[1] = JzAzBz[1];
+  IzAzBz[2] = JzAzBz[2];
+
+  // IzAzBz -> LMS
+  for(int i = 0; i < 3; i++)
+  {
+    LMS[i] = AI[i][0] * IzAzBz[0] + AI[i][1] * IzAzBz[1] + AI[i][2] * IzAzBz[2];
+    LMS[i] = powf(fmax(LMS[i], 0.0f), 1.0f/p);
+    LMS[i] = 10000.f * powf(fmaxf((c1 - LMS[i]) / (c3 * LMS[i] - c2), 0.0f), 1.0f/n);
+  }
+
+  // LMS -> X'Y'Z
+  for(int i = 0; i < 3; i++) XYZ[i] = MI[i][0] * LMS[0] + MI[i][1] * LMS[1] + MI[i][2] * LMS[2];
+
+  // X'Y'Z -> XYZ_D65
+  XYZ_D65[0] = (XYZ[0] + (b - 1.0f) * XYZ[2]) / b;
+  XYZ_D65[1] = (XYZ[1] + (g - 1.0f) * XYZ[0]) / g;
+  XYZ_D65[2] = XYZ[2];
+}
+
```

Comment:
I have checked the maths but not the values in the inverse matrices. I trust that came out of an algebra software ?

Reply:
The matrices have been inverted by octave, e.g. for the M matrix:

```
format long
inv([0.41478972 0.579999 0.014648;-0.20151 1.120649 0.0531008; -0.0166008 0.2648 0.6684799])
```

Reply:
ok, thanks